src/Swift.Bindings/src/Emitter/StringEmitter/Handler/MethodHandler.cs

```diff
@@ -624,130 +691,167 @@ private void EmitSwiftError(IndentedTextWriter writer)
             }
             """;
 
-            writer.WriteLines(text);
-            writer.WriteLine();
+            csWriter.WriteLines(text);
+            csWriter.WriteLine();
         }
 
         /// <summary>
         /// Emits the return statement for the constructor.
         /// </summary>
-        /// <param name="writer">The IndentedTextWriter instance.</param>
-        private void EmitReturnConstructor(IndentedTextWriter writer)
+        /// <param name="csWriter">The IndentedTextWriter instance.</param>
+        private void EmitReturnConstructor(CSharpWriter csWriter, SwiftWriter swiftWriter)
         {
             if (!_requiresIndirectResult)
             {
-                writer.WriteLine("this = result;");
+                csWriter.WriteLine("this = result;");
             }
         }
 
         /// <summary>
         /// Emits the return statement for the method.
         /// </summary>
-        /// <param name="writer">The IndentedTextWriter instance.</param>
-        private void EmitReturnMethod(IndentedTextWriter writer)
+        /// <param name="csWriter">The IndentedTextWriter instance.</param>
+        private void EmitReturnMethod(CSharpWriter csWriter, SwiftWriter swiftWriter)
```

Comment:
This is becoming quite nested as nested in #2955. We should simplify it. 

Can a method be async and return indirect result?

Reply:
Simplified according to https://github.com/dotnet/runtimelab/issues/2955.

> Can a method be async and return indirect result?

No, the return type is always a C# Task.